spec/unit/util/package/version/rpm_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+require 'puppet/util/package/version/rpm'
+
+describe Puppet::Util::Package::Version::Rpm do
+
+  context "when parsing an invalid version" do
+    it "if is parsing symbols" do
+      expect { described_class.parse(:absent)}.to raise_error
+    end
+  end
+
+  context "when creating new version" do
+    it "is parsing basic version" do
+      v = described_class.parse('1:2.8.8-1.el6')
+      expect([v.epoch, v.version, v.release, v.arch ]).to match_array(['1', '2.8.8', '1.el6' , nil])
+    end
+
+    it "is parsing no epoch basic version" do
+      v = described_class.parse('2.8.8-1.el6')
+      expect([v.epoch, v.version, v.release, v.arch ]).to match_array([nil, '2.8.8', '1.el6', nil])
+    end
+
+    it "is parsing no epoch basic short version" do
+      v = described_class.parse('7.15-8.fc29')
+      expect([v.epoch, v.version, v.release, v.arch ]).to match_array([nil, '7.15', '8.fc29', nil])
+    end
+
+    it "is parsing no epoch and no release basic version" do
+      v = described_class.parse('2.8.8')
+      expect([v.epoch, v.version, v.release, v.arch ]).to match_array([nil, '2.8.8', nil, nil])
+    end
+
+    it "is parsing no epoch complex version" do
+      v = described_class.parse('1.4-0.24.20120830CVS.fc31')
+      expect([v.epoch, v.version, v.release, v.arch ]).to match_array([nil, '1.4', '0.24.20120830CVS.fc31', nil])
+    end
+  end
+
+  context "when comparing two versions" do
+    it "epoch has precedence" do
+      first = described_class.parse('0:1.5.3-3.el6')
+      second = described_class.parse('1:1.7.0-15.fc29')
+      expect(first < second).to eql(true)
```

Comment:
I think it is more readable like:
```ruby
expect(first).to be < second
```

https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/comparison-matchers